src/components/AppConfig/HealthCheck.tsx

```diff
@@ -0,0 +1,119 @@
+
+interface HealthCheckDetails {
+  openAI: OpenAIHealthDetails | boolean;
+  vector: VectorHealthDetails | boolean;
+  version: string;
+}
+
+interface OpenAIHealthDetails {
+  configured: boolean;
+  ok: boolean;
+  error?: string;
+  models: Record<string, OpenAIModelHealthDetails>;
+}
+
+interface OpenAIModelHealthDetails {
+  ok: boolean;
+  error?: string;
+}
+
+interface VectorHealthDetails {
+  enabled: boolean;
+  ok: boolean;
+  error?: string;
+}
+
+const isHealthCheckDetails = (obj: unknown): obj is HealthCheckDetails => {
+  return typeof obj === 'object' && obj !== null && 'openAI' in obj && 'vector' in obj && 'version' in obj;
+}
+
+const alertVariants = new Set<AlertVariant>(['success', 'info', 'warning', 'error']);
+const isAlertVariant = (str: string): str is AlertVariant => alertVariants.has(str as AlertVariant);
+const getAlertVariant = (status: string): AlertVariant => {
+  if (status.toLowerCase() === 'ok') {
+    return 'success';
+  }
+  return isAlertVariant(status) ? status : 'info';
+};
+const getAlertSeverity = (status: string, details: HealthCheckDetails): AlertVariant => {
+  const severity = getAlertVariant(status);
+  if (severity !== 'success') {
+    return severity;
+  }
+  if (!isHealthCheckDetails(details)) {
+    return 'success';
+  }
+  if (typeof details.openAI === 'object' && typeof details.vector === 'object') {
+    const vectorOk = !details.vector.enabled || details.vector.ok;
+    return details.openAI.ok && vectorOk ? 'success' : 'warning';
+  }
+  return severity;
+}
+
+export function ShowHealthCheckResult(props: HealthCheckResult) {
+  let severity = getAlertVariant(props.status ?? 'error');
+  if (!isHealthCheckDetails(props.details)) {
+    return (
+      <div className="gf-form-group p-t-2">
+        <Alert severity={severity} title={props.message}>
+        </Alert>
+      </div>
+    );
+  }
+
+  severity = getAlertSeverity(props.status ?? 'error', props.details);
+  const message = severity === 'success' ? 'Health check succeeded!' : props.message;
+
+  return (
+    <div className="gf-form-group p-t-2">
+      <Alert severity={severity} title={message}>
+        <ShowOpenAIHealth openAI={props.details.openAI} />
+        <ShowVectorHealth vector={props.details.vector} />
+      </Alert>
+    </div>
+  );
+}
+
+function ShowOpenAIHealth({ openAI }: { openAI: OpenAIHealthDetails | boolean }) {
+  if (typeof openAI === 'boolean') {
+    return <div>OpenAI: {openAI ? 'Enabled' : 'Disabled'}</div>;
+  }
+  return (
+    <div>
+      <h5>OpenAI</h5>
+      <div>{openAI.ok ? 'OK' : `Error: ${openAI.error}`}</div>
+      <b>Models</b>
+      <table>
+        <thead>
+        </thead>
+        <tbody>
+          {Object.entries(openAI.models).map(([model, details], i) => (
+            <tr key={i}>
+              <td>{model}: </td>
+              <td>{details.ok ? 'OK' : `Error: ${details.error}`}</td>
+            </tr>
+          ))}
+        </tbody>
+      </table>
+    </div>
+  )
```

Comment:
Instead of using a table, what do you think about using just a bullet list with `li`? It looks like a lot of information on a corner.

![Captura de pantalla 2023-10-04 a las 12 25 45](https://github.com/grafana/grafana-llm-app/assets/5699976/5b5b94a5-1344-4d16-8354-30835709017a)

Also, I'd recommend having the banner above the button. You can miss it if you don't scroll.

Reply:
Agreed, I've added that now.